File: src/main/java/leetcode/easy/TwoSum.java
package leetcode.easy;

import java.util.HashMap;
import java.util.Map;

public class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> intMap = new HashMap<>();
        int[] result = new int[2];
        for (int i = 0; i < nums.length; i++) {
            if (intMap.containsKey(target - nums[i])) {
                result = new int[]{intMap.get(target - nums[i]), i};
            }
            intMap.put(nums[i], i);
        }
        return result;
    }
}

File: src/main/java/leetcode/medium/TwoSum.kt
package leetcode.medium

//1. Two Sum
fun main() {
    println(twoSum(intArrayOf(2, 7, 11, 15), 9))
}

fun twoSum(nums: IntArray, target: Int): IntArray {
    val intMap: MutableMap<Int, Int> = HashMap()
    var result = IntArray(2)
    for (i in nums.indices) {
        if (intMap.containsKey(target - nums[i])) {
            result = intArrayOf(intMap[target - nums[i]]!!, i)
        }
        intMap[nums[i]] = i
    }
    return result
}